### PR TITLE
Creating the Objects Pipe and PipeGroup

### LINKED ISSUE #64: Criar objetos PipeGroup e Pipe

### PR DESCRIPTION
**Pull request by edugabriel12**

### CODE REVIEW

File: Prolog/models/pipe.pl [hunk 1]
```diff
@@ -1,3 +1,58 @@
...
+
+toDownString(Pipe, ToDownString):-
+	toUpString(Pipe,ToUpStringPipe),
+	list:join(List, nl, toUpStringPipe),
+	reverse(List, ReversedList),
+	list:join(ReversedList, nl, ToDownString).
+
+toUpString(Pipe, toUpString):-
+	toStringLines(Pipe,StringLines),
+	list:join(StringLines, nl, toUpString).
```
Comment (diego-aquino) [inline Prolog/models/pipe.pl]: Should `toUpString` start with an upper "T" (variable, not atom)?

File: Prolog/models/pipe.pl [hunk 2]
```diff
@@ -1,3 +1,58 @@
...
+	area:createFromString(OriginX, OriginY, String, Area).
+
+toString(Pipe, String):-
+	direction(Pipe, Direction),
+	(Direction = "DOWN" -> toDownString(Pipe, String);
+	toUpString(Pipe,String)).
+
+toDownString(Pipe, ToDownString):-
+	toUpString(Pipe,ToUpStringPipe),
+	list:join(List, nl, toUpStringPipe),
```
Comment (diego-aquino) [inline Prolog/models/pipe.pl]: Did you mean `ToUpStringPipe`?

File: Prolog/models/pipeGroup.pl [hunk 3]
```diff
@@ -1,3 +1,62 @@
...
 
-testMethodPipeGroup(Param):- write(Param).
\ No newline at end of file
+:- use_module(area).
+:- use_module(pipe).
+:- use_module('../utils/list').
+
+create(OriginX, OriginY, TopPipe, BottomPipe, Width, Height, HoleOriginY, HoleHeight, PipeGroup):-
+	TopPipeHeight is HoleOriginY - OriginY,
+	TopPipeHeight is HoleOriginY - OriginY,
```
Comment (diego-aquino) [inline Prolog/models/pipeGroup.pl]: Why are these lines duplicated?

File: Prolog/models/pipeGroup.pl [hunk 4]
```diff
@@ -1,3 +1,62 @@
...
+	BottomPipeHeight is Height - TopPipeHeight - HoleHeight,
+	BottomPipeOriginY is originY + TopPipeHeight + HoleHeight,
```
Comment (diego-aquino) [inline Prolog/models/pipeGroup.pl]: `OriginY`?

File: Prolog/models/pipeGroup.pl [hunk 5]
```diff
@@ -1,3 +1,62 @@
...
+	area:merge(TopPipeArea, BottomPipeArea, Area).
+
+toString(PipeGroup, String):- 
+	topPipe(PipeGroup, TopPipe),
+	bottomPipe(PipeGroup, BottomPipe),
+	holeHeight(PipeGroup, HoleHeight),
+	width(PipeGroup, Width),
+	pipe:toString(TopPipe, TopPipeAsString),
+	pipe:toString(BottomPipe, BottomPipeAsString),
+	list:createList(Width, " ", HoleLine),
```
Comment (diego-aquino) [inline Prolog/models/pipeGroup.pl:58]: This will create a list and unify it to `HoleLine`, but HoleLine should be a string. Did you forget to use join?

**Review by diego-aquino — APPROVED** [review]

**Review by Murilo-Gruppi — APPROVED** [review]

**Review by augustonz — APPROVED** [review]

**Pull request merged**